### PR TITLE
Push decimal

### PR DESCRIPTION
**Pull request by ilovesoup**
In order to make decimal pushable to TiFlash cop, we need to remove checkoverflow and promote_precision. Both TiFlash and TiKV's cop has its own logic of promoting and checkoverflow we should just trust them for their own part.

### CODE REVIEW

**Comment by birdstorm** [comment]
do we also need to remove `MakeDecimal` and `UnscaledValue`?

**Comment by ilovesoup** [comment]
> do we also need to remove `MakeDecimal` and `UnscaledValue`?

when they will be added?

**Comment by birdstorm** [comment]
> > do we also need to remove `MakeDecimal` and `UnscaledValue`?
> 
> when they will be added?

seems that it might be optimized in `sum(some_decimal)` when `some_decimal` can be optimized to `long`?

**Comment by birdstorm** [comment]
I haven't tried it yet but it is stated in logic of `DecimalAggregates`

**Comment by ilovesoup** [comment]
> I haven't tried it yet but it is stated in logic of `DecimalAggregates`

I'd rather first fix as it is. Can you please create a jira issue that MakeDecimal preventing pushdown?

**Comment by birdstorm** [comment]
> > I haven't tried it yet but it is stated in logic of `DecimalAggregates`
> 
> I'd rather first fix as it is. Can you please create a jira issue that MakeDecimal preventing pushdown?

sure.

**Comment by birdstorm** [comment]
TISPARK-420

**Comment by birdstorm** [comment]
should format code using `mvn clean install -DskipTests=true`

File: core/src/main/scala/org/apache/spark/sql/tispark/BasicExpression.scala [hunk 1]
```diff
@@ -130,6 +137,20 @@ object BasicExpression {
       case Like(BasicExpression(lhs), BasicExpression(rhs)) =>
         Some(StringRegExpression.like(lhs, rhs))
 
+      // Coprocessor has its own behavior of type promoting and overflow check
+      // so we simply remove it from expression and let cop handle it
+      case CheckOverflow(BasicExpression(expr), _) =>
```
Comment (zhexuany) [inline core/src/main/scala/org/apache/spark/sql/tispark/BasicExpression.scala]: should we remove MakeDecimal and UnscaledValue?

Comment (birdstorm) [reply core/src/main/scala/org/apache/spark/sql/tispark/BasicExpression.scala]: see discussions above....

Comment (zhexuany) [reply core/src/main/scala/org/apache/spark/sql/tispark/BasicExpression.scala]: ok.

**Pull request merged**